src/libraries/Native/Unix/System.Native/CMakeLists.txt

```diff
@@ -23,6 +23,12 @@ set(NATIVE_SOURCES
     pal_sysctl.c
 )
 
+if (CLR_CMAKE_TARGET_IOS)
+    set(NATIVE_SOURCES ${NATIVE_SOURCES} ios/pal_sys.m)
```

Comment:
I think I'd call this file `pal_log.m`.

We should also not put it in the ios/ directory but rather in the top level as we can use the same implementation for tvOS/watchOS too.